tests/ui/panics/abort-on-panic.rs

```diff
@@ -62,11 +66,19 @@ fn test_always_abort_thread() {
     bomb_out_but_not_abort("joined - but we were supposed to panic!");
 }
 
+fn test_abort_after_transmute() {
+    let unwind: extern "C-unwind" fn() = panic_in_ffi_unwind_abi;
+    let unwind: extern "C" fn() = unsafe { std::mem::transmute(unwind) };
+    std::panic::catch_unwind(|| unwind()).unwrap_err();
```

Comment:
I think this code causes UB, so it shouldn't be added to a run-pass test. We shouldn't have tests with UB.

Reply:
This is UB, unwind across `extern "C"` is not allowed.

Reply:
...ah, okay, I thought we had added a shim such that function _calls_ to `extern "C"` would cause the runtime to abort if the function actually unwinds. I must be misremembering, and indeed the request-for-stabilization says:

> The behavior of function calls using extern "C" is unchanged; thus, it is still undefined behavior to call a C++ function that throws an exception using the extern "C" ABI, even when compiling with panic=unwind.